### PR TITLE
Decoupling metrics from benchmark to allow different kinds of forward process

### PR DESCRIPTION
**Pull request by fengyuentau**
- [x] Decouple metrics from benchmark
- [x] Metric: Base
- [x] Metric: Detection
- [x] Metric: Recognition

This PR is created for the incoming object tracking models and benchmarking.

### CODE REVIEW

**Comment by fengyuentau** [comment]
cc @kaingwade @zihaomu

File: benchmark/config/qrcode_wechatqrcode.yaml [hunk 1]
```diff
@@ -4,18 +4,19 @@ Benchmark:
...
   name: "WeChatQRCode"
-  detect_prototxt_path: "models/qrcode_wechatqrcode/detect_2021sep.prototxt"
-  detect_model_path: "models/qrcode_wechatqrcode/detect_2021sep.caffemodel"
-  sr_prototxt_path: "models/qrcode_wechatqrcode/sr_2021sep.prototxt"
-  sr_model_path: "models/qrcode_wechatqrcode/sr_2021sep.caffemodel"
\ No newline at end of file
+  detect_prototxt_path: "models/qrcode_wechatqrcode/detect_2021nov.prototxt"
+  detect_model_path: "models/qrcode_wechatqrcode/detect_2021nov.caffemodel"
+  sr_prototxt_path: "models/qrcode_wechatqrcode/sr_2021nov.prototxt"
+  sr_model_path: "models/qrcode_wechatqrcode/sr_2021nov.caffemodel"
```
Comment (kaingwade) [inline benchmark/config/qrcode_wechatqrcode.yaml:22]: @fengyuentau how do you define the suffix? Isn't QRCode models trained quite some time ago?

**Comment by fengyuentau** [comment]
The time I upload to zoo. Previous one was wrong.
________________________________
发件人: WU Jia ***@***.***>
发送时间: Monday, November 8, 2021 6:33:43 PM
收件人: opencv/opencv_zoo ***@***.***>
抄送: Yuantao Feng ***@***.***>; Mention ***@***.***>
主题: Re: [opencv/opencv_zoo] Decoupling metrics from benchmark to allow different kinds of forward process (PR #14)


@kaingwade commented on this pull request.

________________________________

In benchmark/config/qrcode_wechatqrcode.yaml<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fopencv%2Fopencv_zoo%2Fpull%2F14%23discussion_r744594257&data=04%7C01%7C%7C190b2ae4e0a54041af9008d9a2a33d41%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637719644252295402%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=MbPkvWM6Uh9RVnsZ16naeLiL6lWtANhtsgVijpki7iM%3D&reserved=0>:

> +  detect_prototxt_path: "models/qrcode_wechatqrcode/detect_2021nov.prototxt"
+  detect_model_path: "models/qrcode_wechatqrcode/detect_2021nov.caffemodel"
+  sr_prototxt_path: "models/qrcode_wechatqrcode/sr_2021nov.prototxt"
+  sr_model_path: "models/qrcode_wechatqrcode/sr_2021nov.caffemodel"


@fengyuentau<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffengyuentau&data=04%7C01%7C%7C190b2ae4e0a54041af9008d9a2a33d41%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637719644252295402%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2BKEMVs%2BrSXOLobcFe4Jsvu97B5itW%2BKPYkXSNA6fRvY%3D&reserved=0> how do you define the suffix? Isn't QRCode models trained quite some time ago?

―
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fopencv%2Fopencv_zoo%2Fpull%2F14%23pullrequestreview-799942663&data=04%7C01%7C%7C190b2ae4e0a54041af9008d9a2a33d41%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637719644252305395%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Ra0NrjZOPxT%2F3N5rduqyWQdcdeUsXgTjw%2FpVh5G2EME%3D&reserved=0>, or unsubscribe<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAEDL63VDMHHBGOG7PT4YFD3UK6RQPANCNFSM5HSB4KQA&data=04%7C01%7C%7C190b2ae4e0a54041af9008d9a2a33d41%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637719644252305395%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=sE68cRfbkMsMmEyXurITAoVXcGmNYbeOzyH1Yw6rR%2FU%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7C%7C190b2ae4e0a54041af9008d9a2a33d41%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637719644252315390%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ZIxnSVh%2Ba420kfIZeFh1BuUZrmmG8jICqKAir4EZ1rg%3D&reserved=0> or Android<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7C%7C190b2ae4e0a54041af9008d9a2a33d41%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637719644252315390%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=u2w3YAxKdp6Irquqndx3yGPe8Mc%2B4w0dN91JC3BI6GY%3D&reserved=0>.

File: benchmark/config/qrcode_wechatqrcode.yaml (hunk 1 above)
Comment (fengyuentau) [reply benchmark/config/qrcode_wechatqrcode.yaml:22]: The suffix is the time I uploaded to zoo. Previous was wrong.

**Pull request merged**